flink-table/flink-table-common/src/main/java/org/apache/flink/table/descriptors/DescriptorProperties.java

```diff
@@ -669,6 +677,14 @@ public DataType getDataType(String key) {
 			}
 		}
 
+		// Extract unique constraints.
+		String pkConstraintNameKey = key + '.' + PRIMARY_KEY_NAME;
+		final Optional<String> pkConstraintNameOpt = optionalGet(pkConstraintNameKey);
+		if (pkConstraintNameOpt.isPresent()) {
+			final String pkColumnsKey = key + '.' + PRIMARY_KEY_COLUMNS;
+			final String columns = optionalGet(pkColumnsKey).orElseThrow(exceptionSupplier(pkColumnsKey));
+			schemaBuilder.primaryKey(pkConstraintNameOpt.get(), columns.split(","));
+		}
```

Comment:
Throw an exception if the the name is not present?

Reply:
I think it is not necessary, the `schema.primary-key.columns` should never appear as a single(e.g. comes from `#putTableSchema`).

Reply:
Then I think we don't need to use `optionalGet` here, we can use `getString` directly.